### ISSUE TITLE
Caprover-Certbot issues with renewing Certificates "certbot renew" error

### ISSUE THREAD

**Issue by fawwal** [caprover/caprover#1790]
**IMPORTANT:** This is a bug report. If you are having problem with deploying a particular app use the deployment issue type.

> If you are having problems with deploying a particular app using a customized dockerfile - please change your questions to Dockerfile and post it where appropriate such as StackOverflow. This is a Docker question, not a CapRover specific issue.

---

What is the problem?
Everything was working perfectly, then my applications certificates expired. I googled around and tried to checked on caprover-certbot, something is wrong here. 

Here's the info I get when I run certbot renew in the container
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2897, in _dep_map
    return self.__dep_map
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2691, in __getattr__
    raise AttributeError(attr)
AttributeError: _DistInfoDistribution__dep_map

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2888, in _parsed_pkg_info
    return self._pkg_info
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2691, in __getattr__
    raise AttributeError(attr)
AttributeError: _pkg_info

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/certbot", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3127, in <module>
    def _initialize_master_working_set():
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3110, in _call_aside
    f(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3139, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 581, in _build_master
    ws.require(__requires__)
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 898, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 792, in resolve
    new_requirements = dist.requires(req.extras)[::-1]
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2635, in requires
    dm = self._dep_map
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2899, in _dep_map
    self.__dep_map = self._compute_dependencies()
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2908, in _compute_dependencies
    for req in self._parsed_pkg_info.get_all('Requires-Dist') or []:
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2890, in _parsed_pkg_info
    metadata = self.get_metadata(self.PKG_INFO)
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 1798, in get_metadata
    raise KeyError("No metadata except PKG-INFO is available")
KeyError: 'No metadata except PKG-INFO is available'
```

Here's a log
![image](https://github.com/caprover/caprover/assets/14047152/388ee5e4-f063-499b-87b5-6bbb3c84fb6a)

Output of the following command on your server:

```
uname -a && lsb_release -a && free -h
```
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.5 LTS
Release:	20.04
Codename:	focal
              total        used        free      shared  buff/cache   available
Mem:          1.9Gi       1.1Gi       145Mi        12Mi       669Mi       631Mi
Swap:         511Mi        52Mi       459Mi

**Comment by githubsaturn** [comment]
Related issues:
https://github.com/caprover/caprover/issues/1310
https://github.com/caprover/caprover/issues/1344
https://github.com/caprover/caprover/issues/1297

What VPS provider are you using?

**Comment by githubsaturn** [comment]
TLDR: very likely a disk issue. Your certbot image may be corrupted.

**Comment by fawwal** [comment]
Using Linode, 

After reading through the issues you linked, thank you. I suspect I bonked my Docker with my ineptitude. I want to blame Linode, but, eh. 

Anyway to delete certbot and rebuild the link, or will I have to start from scratch?

**Comment by githubsaturn** [comment]
Huh! It looks like all these issues are happening on linode... They must be doing something weird... 

Honestly, the easiest way is to use a clean server and move your applications over. 

Otherwise, you can manually pull a fresh image and restart your server and hope that everything works. 

PS: I'm curious, did you do anything specific to your docker?
